fs/config/config.go

```diff
@@ -423,14 +424,31 @@ func SaveConfig() {
 	}
 
 	if err = os.Rename(ConfigPath, ConfigPath+".old"); err != nil && !os.IsNotExist(err) {
-		log.Fatalf("Failed to move previous config to backup location: %v", err)
+		return errors.Errorf("Failed to move previous config to backup location: %v", err)
 	}
 	if err = os.Rename(f.Name(), ConfigPath); err != nil {
-		log.Fatalf("Failed to move newly written config from %s to final location: %v", f.Name(), err)
+		return errors.Errorf("Failed to move newly written config from %s to final location: %v", f.Name(), err)
 	}
 	if err := os.Remove(ConfigPath + ".old"); err != nil && !os.IsNotExist(err) {
 		fs.Errorf(nil, "Failed to remove backup config file: %v", err)
 	}
+	return nil
+}
+
+// SaveConfig calling function which saves configuration file.
+// if saveConfig returns error trying again after sleep.
+func SaveConfig() {
+	var err error
+	for i := 0; i < fs.Config.LowLevelRetries+1; i++ {
+		if err = saveConfig(); err == nil {
+			return
+		}
+		waitingTimeMs := mathrand.Intn(1000)
+		time.Sleep(time.Duration(waitingTimeMs) * time.Millisecond)
+	}
+	log.Fatalf(err.Error())
```

Comment:
Using `log.Fatalf` means that if the error message contains `%` characters then `log.Fatalf` will try to interpolate them.

So you can either just use `log.Fatal(err)` or you could use `log.Fatalf("Failed to save config after %d tries: %v", fs.Config.LowLevelRetries, err)`